packages/typespec-client-generator-core/src/decorators.ts

```diff
@@ -1190,6 +1191,83 @@ export function getClientLocation(
     | undefined;
 }
 
+const legacyHierarchyBuildingKey = createStateSymbol("legacyHierarchyBuilding");
+
+function isPropertySuperset(target: Model, value: Model): boolean {
```

Comment:
If is possible to use `sourceProperty` along with `sourceModel` to handle comparison logic?